app/models/order.rb

```diff
@@ -4,6 +4,7 @@ class Order < ApplicationRecord
   has_many :order_details, dependent: :destroy
   delegate :name, to: :user, prefix: true
   delegate :phone, :address_detail, to: :address, prefix: true
+  before_save :restore_product_quantity, if: :check_status
```

Comment:
```suggestion
  before_save :restore_product_quantity, if: ->{sefl.rejected?}
```
ri dc nhé
tận dụng thêm method của enum hỗ trợ nè 
ngoài ra  nếu xử lý ngắn gọn và ko cần tái sử dụng thì tận dụng bên trên thay vì tạo method nhé